src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.iOS.cs

```diff
@@ -51,62 +42,19 @@ private bool LoadCalendarDataFromNative(string localeName, CalendarId calendarId
                 saMonthNames[6] = leapHebrewMonthName;
 
             }
-            saAbbrevMonthNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthNames).Split("||"), calendarId, ref leapHebrewMonthName);
-            saMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.MonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
-            saAbbrevMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            EnumMonthNames(localeName, calendarId, CalendarDataType.AbbrevMonthNames, out this.saAbbrevMonthNames!, ref leapHebrewMonthName);
+            EnumMonthNames(localeName, calendarId, CalendarDataType.MonthGenitiveNames, out this.saMonthGenitiveNames!, ref leapHebrewMonthName);
+            EnumMonthNames(localeName, calendarId, CalendarDataType.AbbrevMonthGenitiveNames, out this.saAbbrevMonthGenitiveNames!, ref leapHebrewMonthName);
 
-            saEraNames = NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.EraNames).Split("||"));
-            saAbbrevEraNames = Array.Empty<string>();
+            EnumEraNames(localeName, calendarId, CalendarDataType.EraNames, out this.saEraNames!);
+            EnumEraNames(localeName, calendarId, CalendarDataType.AbbrevEraNames, out this.saAbbrevEraNames!);
 
             return sNativeName != null && saShortDates != null && saLongDates != null && saYearMonths != null &&
                    saDayNames != null && saAbbrevDayNames != null && saSuperShortDayNames != null && saMonthNames != null &&
                    saAbbrevMonthNames != null && saMonthGenitiveNames != null && saAbbrevMonthGenitiveNames != null &&
                    saEraNames != null && saAbbrevEraNames != null;
```

Comment:
Let's refactor this also to have like in .Icu files, as now most of the functions we call return boolean we don't need anymore to check for all these values.

Reply:
We still have NativeName and MonthDay that use the old style. Would you prefer to refactor those + the return in this PR or leave it for the future if we decide to completely eliminate the `CalendarData.iOS.cs`?

Reply:
Let's handle those changes in a separate PR.